src/transformStateWithClones.js

```diff
@@ -6,8 +6,51 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  // const newObject = {};
+
+  const clone = {}; // новий порожній об'єкт
+  const resultArray = [];
+
+  for (const key in state) {
+    clone[key] = state[key];
+  }
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        addProperties(clone, action.extraData);
+        break;
+
+      case 'removeProperties':
+        removeProperties(clone, action.keysToRemove);
+        break;
+
+      case 'clear':
+        clearProperties(clone);
+        break;
+    }
```

Comment:
The switch statement is missing a default case. It's a good practice to have a default case for error handling, even if it's just to throw an error or to handle unexpected action types.